stdlib/bytes.mli

```diff
@@ -216,6 +216,39 @@ val escaped : bytes -> bytes
     @raise Invalid_argument if the result is longer than
     {!Sys.max_string_length} bytes. *)
 
+val escaped_ascii : bytes -> bytes
+(** [escape_ascii s] is [s] with all characters outside the printable US-ASCII
+    range represented by escape sequences.
+
+    All characters outside the US-ASCII printable range \[0x20 .. 0x7E\] are
+    escaped, as well as backslash (0x2F) and double-quote (0x22).
+
+    The function {!Scanf.unescaped} is a left inverse of [escaped],
+    i.e. [Scanf.unescaped (escaped s) = s] for any string [s] (unless
+    [escaped s] fails).
+
+    @raise Invalid_argument if the result is longer than
+    {!Sys.max_string_length} bytes.
+
+    @since 4.14.0 *)
+
+val escaped_utf8 : bytes -> bytes
```

Comment:
This should be `utf_8` for consistency with `Buffer.add_uchar_utf_8`. 

Personally I don't think `String.escaped_{ascii,utf_8}` reads well. From the name I would rather expect them to return a string in which ascii or UTF-8 is escaped. Maybe `String.escaped_for_{ascii,utf_8}` ?

Reply:
I fear we have to choose what exactly we make it consistent with, unless we can agree to revamp the other functions. I was aiming for consistency with the rest of the `Bytes/String` module where we have functions like `capitalize_ascii` and `lowercase_ascii` and where `_ascii` essentially means "assume all characters are ASCII".

Reply:
> like `capitalize_ascii` and `lowercase_ascii` and where `_ascii` essentially means "assume all characters are ASCII".

Personally, that's not the way I read these functions, I read these as: "capitalize ASCII characters". 

I don't think there's any need to revamp other functions or make these ones read wrong in the name of consistency. `String.escaped_ascii` really reads to me as a "string with ASCII characters escaped" which is really not what it does.

Reply:
Myself, I definitely agree with the idea of hex escapes, and if I were to design the interface, I'd also add an option to exclude certain characters from escaping (if you are formatting strings for something other than OCaml, that becomes relevant). However, I don't know how well it aligns with maintainers' plans.

Anyway, I fixed that sleep deprivation induced blunder that broke the tests, and it may be a good idea to collect all questions about the interface and proposed options now to make it easier to come to a consensus.

Reply:
If you ask me, I don't find the function added by this module very useful and pertinent. 

I rarely need to escape my bytes according to OCaml syntax. Often it's just quotes or double quotes. 

I would much prefer a little infrastructure to easily devise byte escaping functions. For example something [like that](https://erratique.ch/software/b0/doc/B00_std/String/index.html#escunesc).

Reply:
From my point of view, the `escaped_for_utf_8` (or `utf_8_escaped`?) function is just an updated and improved version of the `escaped` function which was already used in the toplevel. It is indeed quite specific to the `OCaml compiler` itself.